src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -0,0 +1,1845 @@
+
+        [Theory]
+        [MemberData(nameof(GetMethodGroupImplicitConversionData))]
+        public void MethodGroup_ImplicitConversion(string methodDeclaration, string methodGroupExpression, DiagnosticDescription[]? expectedDiagnostics, string? expectedType)
+        {
+            var source =
+$@"class Program
+{{
+    {methodDeclaration}
+    static void Main()
+    {{
+        System.Delegate d = {methodGroupExpression};
+        System.Console.Write(d.GetDelegateTypeName());
+    }}
+}}";
+            var comp = CreateCompilation(new[] { source, s_utils }, parseOptions: TestOptions.RegularPreview, options: TestOptions.ReleaseExe);
+            if (expectedDiagnostics is null)
+            {
+                CompileAndVerify(comp, expectedOutput: expectedType);
+            }
+            else
+            {
+                comp.VerifyDiagnostics(expectedDiagnostics);
+            }
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var expr = tree.GetRoot().DescendantNodes().OfType<VariableDeclaratorSyntax>().Single().Initializer!.Value;
+            var typeInfo = model.GetTypeInfo(expr);
+            Assert.Null(typeInfo.Type);
+            Assert.Equal(SpecialType.System_Delegate, typeInfo.ConvertedType!.SpecialType);
+        }
+
+        public static IEnumerable<object?[]> GetMethodGroupExplicitConversionData()
+        {
+            return GetMethodGroupData((methodGroupExpression, methodGroupOnly) =>
+                {
+                    int offset = methodGroupExpression.Length - methodGroupOnly.Length;
+                    return new[]
+                        {
+                            // (6,20): error CS0030: Cannot convert type 'method' to 'Delegate'
+                            //         object o = (System.Delegate)F;
+                            Diagnostic(ErrorCode.ERR_NoExplicitConv, $"(System.Delegate){methodGroupExpression}").WithArguments("method", "System.Delegate").WithLocation(6, 20)
+                        };
+                });
+        }
+
+        [Theory]
+        [MemberData(nameof(GetMethodGroupExplicitConversionData))]
+        public void MethodGroup_ExplicitConversion(string methodDeclaration, string methodGroupExpression, DiagnosticDescription[]? expectedDiagnostics, string? expectedType)
+        {
+            var source =
+$@"class Program
+{{
+    {methodDeclaration}
+    static void Main()
+    {{
+        object o = (System.Delegate){methodGroupExpression};
+        System.Console.Write(o.GetType().GetTypeName());
+    }}
+}}";
+            var comp = CreateCompilation(new[] { source, s_utils }, parseOptions: TestOptions.RegularPreview, options: TestOptions.ReleaseExe);
+            if (expectedDiagnostics is null)
+            {
+                CompileAndVerify(comp, expectedOutput: expectedType);
+            }
+            else
+            {
+                comp.VerifyDiagnostics(expectedDiagnostics);
+            }
```

Comment:
Consider testing the type/converted type from the semantic model here as well.

Reply:
I meant of the method group itself, not the explicit conversion on top :)

---
In reply to: [618739761](https://github.com/dotnet/roslyn/pull/52448#discussion_r618739761) [](ancestors = 618739761)